### DISCUSSION TITLE
Actions in this workflow must be: within a repository owned by

### DISCUSSION THREAD

**Discussion by David-Lor** [community/community#26609] [Actions]
<p>I’ve tried changing the Actions settings in one of my repositories, to “Allow local actions only”. However, an action I have on it, which is scheduled to run daily, failed with the error: “<em>actions/checkout@master, actions/setup-python@master, actions/checkout@master, actions/setup-python@master, and docker/login-action@v1 are not allowed to be used in David-Lor/Docker-Python-Git-App. Actions in this workflow must be: within a repository owned by David-Lor</em>” (links to <a href="https://github.com/David-Lor/Docker-Python-Git-App/blob/9470289698d08fe70a773b3c6bad32b8170229d7/.github/workflows/build_test_push.yaml" rel="noopener nofollow ugc">workflow file</a> and <a href="https://github.com/David-Lor/Docker-Python-Git-App/actions/runs/731441088" rel="noopener nofollow ugc">failed run</a>).</p>
<p>From what I understand reading the description of the setting (“Only actions defined in a repository within David-Lor can be used.”), the action should still be running, since it’s in my own repository.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Only the <em>workflow</em> is defined in your repository. The actions listed in the error messages are owned by the <a href="https://github.com/actions">actions</a> and <a href="https://github.com/docker">docker</a> organizations. You can see that by the part in front of the first slash, e.g. <a href="https://github.com/actions/checkout"><code>actions/checkout</code></a> refers to a repository that’s definitely not owned by you. :wink:</p>

**Comment by David-Lor** [comment]
<p>Oh I see! it makes sense. The naming can be confusing sometimes :sweat_smile:</p>

**Comment by mouse07410** [comment]
What then is the remedy? If the repository is owned by me, but docker and actions aren't? Does it mean I cannot use checkout, setup, and cache?

```
[Error: .github#L1](https://github.com/mouse07410/stepic-tests/commit/abf540ad0a173e6bf5688e5d4f1f15af531b245a#annotation_9966423424)
actions/checkout@v3, haskell/actions/setup@v2, and actions/cache@v3 are not allowed to be used in mouse07410/stepic-tests. Actions in this workflow must be: within a repository owned by mouse07410.
```

**Reply by airtower-luna** [reply]
The reasonable thing would be to allow those actions in the repository settings (under Actions), instead of allowing only actions that belong to your account.

Alternatively you can fork the actions' repositories into your account and use them from there, but then you'll have to maintain the forks forever – and most likely by just pulling upstream changes. :grin:

**Comment by mouse07410** [comment]
> The reasonable thing would be to allow those actions in the repository settings (under Actions), instead of allowing only actions that belong to your account

I am not sure I understand - this *is* my repository created by me in my account. 
Does this not mean that this action *is* allowed in the repository?
 
<img width="1293" alt="Screenshot 2023-03-24 at 07 05 50" src="https://user-images.githubusercontent.com/5923577/227505170-7c167e86-39f3-4bb0-a45e-36c57c91d157.png">

Also, it worked fine up until last month. If it's disabled somewhere at the account level - could you please point me at the location of that setting?

Here's the Plan summary - I'm staying on the Free Tier:
<img width="720" alt="Screenshot 2023-03-24 at 07 18 53" src="https://user-images.githubusercontent.com/5923577/227507805-d1adf2ac-2553-4543-9899-4b575d1799ec.png">

<img width="712" alt="Screenshot 2023-03-24 at 07 20 06" src="https://user-images.githubusercontent.com/5923577/227508043-91f9c578-d3db-4fcb-8832-cf5928940431.png">

I only see "Prebuilt storage" as red-lining. How do I find and clear or eliminate it?

> Alternatively you can fork the actions' repositories into your account and use them from there

Again, I don't understand - it's *my* repository. There is no upstream.

**Comment by airtower-luna** [comment]
@mouse07410, the error message shows you're using 3 actions that aren't yours: `actions/checkout@v3, haskell/actions/setup@v2, and actions/cache@v3`

The repositories they come from belong to the `actions` and `haskell` organizations. The error message says you have prohibited the repository from running actions that don't belong to you, which includes those three. To allow them to be used you can either allow them specifically, or others' actions in general.

**Comment by mouse07410** [comment]
> The repositories they come from belong to the actions and haskell organizations. The error message says you have prohibited the repository from running actions that don't belong to you, which includes those three. To allow them to be used you can either allow them specifically, or others' actions in general.

Could you please guide me? How can I allow them? What do I need to do? 
I don't recall prohibiting actions, and, as I said, it used to work... Where do I look? Thanks!

**Reply by airtower-luna** [reply]
Check the repository settings, and select "Actions > General" in the menu. There you can choose which (if any) actions to allow. The documentation is here: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository

**Comment by mouse07410** [comment]
**Thank you!** Much better! From three, now the problem is reduced to one action:
```
[Error: .github#L1](https://github.com/mouse07410/stepic-tests/commit/ba417ab1113043703abbbc5adde3c8d9a9a69270#annotation_9991190358)
haskell/actions/setup@v2 is not allowed to be used in mouse07410/stepic-tests. Actions in this workflow must be: within a repository owned by mouse07410 or created by GitHub.
```

Do you see any problem with this configuration?
<img width="715" alt="Screenshot 2023-03-24 at 13 35 46" src="https://user-images.githubusercontent.com/5923577/227599482-132027f9-8e87-4457-8c00-17379f7279cc.png">

**Reply by airtower-luna** [reply]
Yes, you're still not allowing actions from the `haskell` organization (`actions` is from GitHub). What's odd is that in the screenshot you don't have a field to enter patterns of actions to allow. Here's what it that settings section looks like for me:

![grafik](https://user-images.githubusercontent.com/5814919/227606024-c3ccee73-ca5f-4caa-b4dc-0c8f83d70aaa.png)

Check if you have any browser setting, add-on, or similar that might hide the text field, and if not you might want to send a bug report to GitHub support.

**Comment by mouse07410** [comment]
> What's odd is that in the screenshot you don't have a field to enter patterns of actions to allow

Yeah - your screenshot has it, and none of my browsers seems to... Tried on Chrome and Safari - neither one shows that "Allow specified actions" text field.

> Check if you have any browser setting, add-on, or similar that might hide the text field, 

Checked - can't see anything that could possibly interfere. 

> you might want to send a bug report to GitHub support

Yeah... But they did not make it easy at all. Here are the options:
<img width="1025" alt="Screenshot 2023-03-24 at 15 42 34" src="https://user-images.githubusercontent.com/5923577/227624097-76229919-3b55-49a1-b51a-d3ccb4df3769.png">

It seems to force me to "GitHub Community". I may be dense today, but I don't see how to get the bug report to GitHub support. :-(

**Comment by airtower-luna** [comment]
That's odd, I do have a "report a bug" option on that page. I wonder if the direct link works for you? :sweat_smile: https://support.github.com/contact/bug-report

Meanwhile you can use the "allow all actions and reusable workflows" option to disable limits on what you can call with `uses` in your repository.

**Comment by mouse07410** [comment]
Thank you - opened ticket https://support.github.com/ticket/personal/0/2077502

In the meanwhile, as a workaround I had to relax to "Allow all actions and reusable workflows". Not sure what the security consequences are...

Also, an unrelated question: it looks like I've exhausted "Actions and Packages Storage" - how do I (a) find out what actions or packages consumed it, and (b) clear it all out? Thanks!

**Comment by airtower-luna** [comment]
> In the meanwhile, as a workaround I had to relax to "Allow all actions and reusable workflows". Not sure what the security consequences are...

For a personal repository, probably none at all. That setting simply restricts what you can call with `uses:` in your workflows. If only you can commit and you review any PR properly, you have the same degree of control. It's much more interesting if you have others with direct commit access and you want to enforce a policy on what may or may not be used.

> Also, an unrelated question: it looks like I've exhausted "Actions and Packages Storage" - how do I (a) find out what actions or packages consumed it, and (b) clear it all out? Thanks!

My first suspicion would be artifacts. Also in the billing section of your settings you can get a usage report, maybe that helps, not sure how detailed it is.

**Comment by mouse07410** [comment]
> My first suspicion would be artifacts. Also in the billing section of your settings you can get a usage report, maybe that helps, not sure how detailed it is.

I think you're correct. It says the space is taken by prebuilt's. Only question - there are four pages of prebuilt in one project only (the one we're talking about). I don't think I have prebuilt in other projects, but did not check them all.

What would be a good way to delete all of the old prebuilt's? There's no way provided in the billing section, and I saw no way on the repo page. After I've deleted several prebuilt's manually, there was no increase of free space - still shows as 100% used up...
